src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentsService.kt

```diff
@@ -63,9 +65,16 @@ class AppointmentsService(
   ): ActionPlanAppointment {
     val appointment = getActionPlanAppointmentOrThrowException(actionPlanId, sessionNumber)
     setAttendanceFields(appointment, attended, additionalInformation)
+    if (attended == Attended.NO) {
+      notifyNoAttendance(appointment)
+    }
     return actionPlanAppointmentRepository.save(appointment)
   }
 
+  private fun notifyNoAttendance(appointment: ActionPlanAppointment) {
```

Comment:
does this need to be in its own method?

Reply:
Perhaps the check for non-attendance should be in the method and the method called notifyIfNonAttendance which means the main method just orchestrates

Reply:
👍  done